pkg/i2gw/ingress2gateway.go

```diff
@@ -120,16 +123,19 @@ func ConstructIngressesFromCluster(ctx context.Context, cl client.Client, ingres
 
 // constructProviders constructs a map of concrete Provider implementations
 // by their ProviderName.
-//
-// TODO (levikobi): Issue #45 - let users filter by provider name.
-func constructProviders(conf *ProviderConf) map[ProviderName]Provider {
+func constructProviders(conf *ProviderConf, providers []string) (map[ProviderName]Provider, error) {
```

Comment:
`ProviderConf` should not be per provider? I know no one is using it now but IIRC that was the purpose of it.

Reply:
I think `ProviderConf` can be shared between all providers, at least for now.
If any provider needs unique settings, it can instantiate them itself.

At some point, we might want to push specific user configurations for a specific provider - then we'll have to rethink it.
But again, for now, IMO it's good enough. Curious what's your opinion about it @LiorLieberman

Reply:
sgtm if you are good with it.